hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -281,7 +281,7 @@ class HoodieMergeOnReadRDD(@transient sc: SparkContext,
         tableState.requiredStructSchema.foreach(
           f => {
             val curPos = posIterator.next()
-            val curField = row.get(curPos, f.dataType)
+            val curField = if (row.isNullAt(curPos)) null else row.get(curPos, f.dataType)
```

Comment:
Hi @Liulietong , thanks for your contribution! Is that possible to add a test case to verify this will fix the vectorized reader bug?